Anature/src/application/controllers/BattleController.java

```diff
@@ -1258,6 +1319,137 @@ private void onBackBtn()
 		mShowMoveSe.set(false);
 		mShowBtns.set(true);
 	}
+	
+	private void statusEffectForTurn(Anature anature, boolean isPlayer, int indexOfMove) {
+		StatusEffects anatureStatus = anature.getStatus();
+		
+		switch (anatureStatus) {
+			case Sleep:
+				mClickQueue.enqueueToFront(new Runnable() {
+					
+					@Override
+					public void run() {
+						if(isPlayer) {
+							mFightManager.attackEnemy(-1);
+						} else {
+							mFightManager.attackPlayer(-1);
+						}
+					}
+				});
+
+				mClickQueue.enqueueToFront(new Runnable() {
+	
+					@Override
+					public void run() {
+						if(isPlayer) {
+							mStatusIconEnemy.setImage(mSleepStatusIcon);
+						} else {
+							mStatusIconPlayer.setImage(mSleepStatusIcon);
+						}
+						
+						mDialogueTxt.set(anature.getName() +  " is fast asleep!");
+					}
+				});
+				break;
+	
+			case Paralysis:
+				
+				mClickQueue.enqueueToFront(new Runnable() {
+					
+					@Override
+					public void run() {
+						if(Math.random() > 0.25) {
+								healthDrainMove(isPlayer ? mFightManager.attackEnemy(indexOfMove) : mFightManager.attackPlayer(indexOfMove),
+												isPlayer ? mPlayerHp : mEnemyHp);
+							
+						} else {
+							if(isPlayer) {
+								mFightManager.attackEnemy(-1); 
+							} else {
+								mFightManager.attackPlayer(-1); 
+							}
+						}
+					}
+				});
+				
+				mClickQueue.enqueueToFront(new Runnable() {
+	
+					@Override
+					public void run() {
+						if(isPlayer) {
+							mStatusIconEnemy.setImage(mParalyzedStatusIcon);
+						} else {
+							mStatusIconPlayer.setImage(mParalyzedStatusIcon);
+						}
+						
+						mDialogueTxt.set(anature.getName() +  " is paralysed! It may not be able to move!");
+						
+					}
+				});
+				
+				break;
+				
+			case Burn:
+				mClickQueue.enqueueToFront(new Runnable() {
+	
+					@Override
+					public void run() {
+						
+						if(isPlayer) {
+							healthDrainMove(mFightManager.attackEnemy(indexOfMove),
+									mEnemyHp);
+						} else {
+							healthDrainMove(mFightManager.attackPlayer(indexOfMove),
+									mPlayerHp);
+						}
+					}
+					
+				});
+				
+				mClickQueue.enqueueToFront(new Runnable() {
+					
+					@Override
+					public void run() {
+						
+							healthDrainStatus(anature.getName() + "is hurt because it is burned!", 
+									(double) anature.getTotalHp()/16, 
+									isPlayer);
+					}
+					
+				});
+				
+				mClickQueue.enqueueToFront(new Runnable() {
```

Comment:
Should check if the Image was already set to burn. If it hasn't been, then set it and display the text, otherwise just ignore it since the user doesn't need to be informed of a burn status every single turn.